### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by arnopo**
Update CI to build the https://github.com/OpenAMP/openamp-system-reference/tree/main/examples/legacy_apps applications instead of the https://github.com/OpenAMP/open-amp/tree/main/apps applications, which will be deprecated in the next release.

### CODE REVIEW

**Review by tnmysh — APPROVED** [review]
Looks good to me.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -23,12 +23,17 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout open-amp
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
```
Comment (wmamills) [inline .github/workflows/continuous-integration.yml:26]: "path: open-amp" HERE ???
with adjustments in script?

I think libmetal and openamp-system-reference are sub directories of open-amp.  That works but is a little strange.  Is this easy to fix as described above??  If not we will deal with it later.

**Review by wmamills — CHANGES_REQUESTED** [review]
In general looks good, some nits you should fix.
And then decide to tackle the dir structure or not

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (wmamills) [reply .github/workflows/continuous-integration.yml:26]: I take this comment back.  Every build would need to be updated not just the linux build.
We should do this in another PR

**Review by wmamills — APPROVED** [review]
I reviewed this and tested it as well.
I think it is ready.

**Pull request merged**